src/machines.rs

```diff
@@ -0,0 +1,111 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use tokio::sync::RwLock;
+use tracing::{debug, error};
+
+use crate::MachineStats;
+use crate::MonitordStats;
+
+pub async fn get_machines(
+    connection: &zbus::Connection,
+    config: &crate::config::Config,
+) -> Result<HashMap<String, u32>, zbus::Error> {
+    let c = crate::dbus::zbus_machines::ManagerProxy::new(connection).await?;
+    let mut results = HashMap::<String, u32>::new();
+
+    let machines = c.list_machines().await?;
+
+    for machine in machines.iter().filter(|c| &c.1 == "container") {
+        let m = c.get_machine(&machine.0).await?;
+        let leader = m.leader().await?;
+
+        if config.machines.auto_discover || config.machines.machines_list.contains(&machine.0) {
+            results.insert(machine.0.to_string(), leader);
+        }
+    }
+
+    return Ok(results);
+}
+pub async fn update_machines_stats(
+    config: crate::config::Config,
+    connection: zbus::Connection,
+    locked_monitord_stats: Arc<RwLock<MonitordStats>>,
+) -> anyhow::Result<()> {
+    let locked_machine_stats: Arc<RwLock<MachineStats>> =
+        Arc::new(RwLock::new(MachineStats::default()));
+
+    for (machine, leader) in get_machines(&connection, &config).await?.into_iter() {
```

Comment:
Took me a bit to understand this / what it is - Does this make it clearer?

Ignore me here if `'leader'` is accepted terminology in systemd machine terms and documentation. Would be my lack of understanding if so.
```suggestion
    for (machine, leader_pid) in get_machines(&connection, &config).await?.into_iter() {
```